src/api/app/components/workflow_run_detail_component.rb

```diff
@@ -22,4 +23,21 @@ def parse_payload(workflow_run)
   rescue JSON::ParserError
     workflow_run.request_payload
   end
+
+  def workflow_configuration_available?
+    workflow_run.workflow_configuration_url || workflow_run.workflow_configuration_path
+  end
+
+  # Old workflow run entries didn't store the configuration-related information
+  def workflow_configuration_data
```

Comment:
@saraycp why not doing this in the `haml` file of the view component?

Reply:
IMO it would be nice to read

Reply:
TBH, because haml-lint complained about every change I tried.

Do you remember the error you had some time ago? It happened to me again.

```
rails dev:lint:all                                                                                                                                                             
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem                                                                                                                                                                                                                                                      
Running haml-lint in src/api...                                                                                                                                                                                    
/usr/lib64/ruby/gems/3.1.0/gems/parallel-1.23.0/lib/parallel.rb:606:in `dump': can't dump hash with default proc (TypeError)                                                                                       
        from /usr/lib64/ruby/gems/3.1.0/gems/parallel-1.23.0/lib/parallel.rb:606:in `process_incoming_jobs'                                                                                                        
        from /usr/lib64/ruby/gems/3.1.0/gems/parallel-1.23.0/lib/parallel.rb:577:in `block in worker'                                                                                                              
        from /usr/lib64/ruby/gems/3.1.0/gems/parallel-1.23.0/lib/parallel.rb:568:in `fork'                                                                                                                         
        from /usr/lib64/ruby/gems/3.1.0/gems/parallel-1.23.0/lib/parallel.rb:568:in `worker'                                                                                                                                                                                                                                 
...                                                                               
        from /usr/lib64/ruby/gems/3.1.0/gems/railties-7.0.4.3/lib/rails/commands/rake/rake_command.rb:18:in `perform'                                                                         
        from /usr/lib64/ruby/gems/3.1.0/gems/railties-7.0.4.3/lib/rails/command.rb:51:in `invoke'                                                                                             
        from /usr/lib64/ruby/gems/3.1.0/gems/railties-7.0.4.3/lib/rails/commands.rb:18:in `<top (required)>'                                                                                  
        from bin/rails:4:in `require'                                                          
        from bin/rails:4:in `<main>'                                                           
marshal data too short                                                                         
<internal:marshal>:34:in `load'                                                                
/usr/lib64/ruby/gems/3.1.0/gems/parallel-1.23.0/lib/parallel.rb:81:in `work'                   
/usr/lib64/ruby/gems/3.1.0/gems/parallel-1.23.0/lib/parallel.rb:522:in `block (4 levels) in work_in_processes'                                                                                
/usr/lib64/ruby/gems/3.1.0/gems/parallel-1.23.0/lib/parallel.rb:637:in `with_instrumentation'  
/usr/lib64/ruby/gems/3.1.0/gems/parallel-1.23.0/lib/parallel.rb:521:in `block (3 levels) in work_in_processes'                                                                                
/usr/lib64/ruby/gems/3.1.0/gems/parallel-1.23.0/lib/parallel.rb:509:in `loop'                  
/usr/lib64/ruby/gems/3.1.0/gems/parallel-1.23.0/lib/parallel.rb:509:in `block (2 levels) in work_in_processes'                                                                                
/usr/lib64/ruby/gems/3.1.0/gems/parallel-1.23.0/lib/parallel.rb:219:in `block (4 levels) in in_threads' 
```

Reply:
Moreover, writing it in Haml implies a lot of embedded Ruby code in the view (`- if`, `- else`, `= <ruby code>`...).
Something that ViewComponents are meant to avoid: https://viewcomponent.org/#why-use-viewcomponents